app/models/pipeline_run.rb

```diff
@@ -607,24 +606,25 @@ def load_taxons(downloaded_json_path, refined = false)
     return if loaded_records > 0
 
     # only keep counts at certain taxonomic levels
-    taxon_counts_attributes_filtered = []
     acceptable_tax_levels = [TaxonCount::TAX_LEVEL_SPECIES]
     acceptable_tax_levels << TaxonCount::TAX_LEVEL_GENUS if multihit?
     acceptable_tax_levels << TaxonCount::TAX_LEVEL_FAMILY if multihit?
-    pipeline_output_dict['taxon_counts_attributes'].each do |tcnt|
+    taxon_counts_attributes_filtered = taxon_counts_attributes.select do |tcnt|
       # TODO:  Better family support.
-      if acceptable_tax_levels.include?(tcnt['tax_level'].to_i) && !invalid_family_call?(tcnt)
-        taxon_counts_attributes_filtered << tcnt
-      end
+      acceptable_tax_levels.include?(tcnt['tax_level'].to_i) && !invalid_family_call?(tcnt)
     end
     # Set created_at and updated_at
     current_time = Time.now.utc # to match TaxonLineage date range comparison
+    tcnt_attrs_to_merge = {
+      'created_at' => current_time,
+      'updated_at' => current_time,
+      'pipeline_run_id' => id
+    }
     taxon_counts_attributes_filtered.each do |tcnt|
-      tcnt["created_at"] = current_time
-      tcnt["updated_at"] = current_time
       tcnt["count_type"] += "+" if refined
+      tcnt.merge!(tcnt_attrs_to_merge)
     end
-    update(taxon_counts_attributes: taxon_counts_attributes_filtered)
+    TaxonCount.import!(taxon_counts_attributes_filtered)
```

Comment:
If we're concerned about duplicate insertions, should we specify ignoring duplicates instead of throwing an exception? https://github.com/zdennis/activerecord-import#duplicate-key-ignore

Reply:
I saw that option, but haven't used that on intent. We shouldn't have this race condition. I want to leave this command blowing up in case that happens, because that means we need to fix something else.

Reply:
This line (pre-existing in the code) should prevent this situation:
https://github.com/chanzuckerberg/idseq-web/blob/bd727a81cfac4eb6ee7a3edafb18b28ef449d1ec/app/models/pipeline_run.rb#L606

Unless we are processing the same pipeline_run_id at the same time, which is exactly one of my hypothesis for the original issue.